File: backend/src/main/java/com/backend/ord/services/UserSessionService.java
package com.backend.ord.services;

import com.backend.ord.domain.entities.UserSession;
import com.backend.ord.exceptions.UserNotFoundException;

import java.util.Optional;
import java.util.UUID;

public interface UserSessionService {
    UserSession save (UserSession userSession);

    Optional<UserSession> findByToken(String token);

    Optional<UserSession> findByTokenAndUserId(String token, UUID userId);

    void openSessionFromJWT(String token) throws UserNotFoundException;

    void deleteSessionByToken(String authToken);
}

File: backend/src/main/kotlin/com/backend/ord/services/UserSessionService.kt
package com.backend.ord.services

import com.backend.ord.core.auth.models.UserSessionEntity
import com.backend.ord.exceptions.UserNotFoundException
import java.util.*

interface UserSessionService {
    fun save(userSession: UserSessionEntity): UserSessionEntity

    fun findByToken(token: String): UserSessionEntity?

    fun findByTokenAndUserId(token: String, userId: UUID): UserSessionEntity?

    @Throws(UserNotFoundException::class)
    fun openSessionFromJWT(token: String)

    fun deleteSessionByToken(authToken: String)
}
